slack_today_i_did/our_repo.py

```diff
@@ -147,10 +180,7 @@ def what_kinda_file(self, filename: str) -> ElmVersion:
         with open(filename) as f:
             for line in f:
                 if line.strip():
-                    if 'exposing' in line:
-                        self._known_files[ElmVersion.v_017].append(filename)
-                        return ElmVersion.v_017
-                    if 'where' in line:
-                        self._known_files[ElmVersion.v_016].append(filename)
-                        return ElmVersion.v_016
+                    elm_version = self._elm_version_text(line)
+                    self._known_files[elm_version].append(filename)
+
```

Comment:
:bug: what_kinda_file isn't returning at the earliest possible chance anymore and will classify any file as unknown in the end, won't it. how about extracting 

```python
def _detect_elm_version(self, filename):
    with open(filename) as f:
        for line in f:
            ...
```
? also, tests would be nice